### PR TITLE
Additional arithmetic ops/traits implementations for Array and &Array

### LINKED ISSUE #103: Implement arithmetic ops for &Array vs &Array
Is there any reason why arithmetic overloading only exists for `Array`-`Array`? Given that the `arrayfire::add/sub/mul` functions all take `&Array` arguments it seems like it is worth having the extra syntactic sugar.

Just to be clear, I mean implementations like this:

```rust
impl<'a> Add<&'a Array> for Array { }
impl<'a> Add<Array> for &'a Array { }
impl<'a, 'b> Add<&'a Array> for &'b Array { }
```

Internally these would just call `arrayfire::add`.

### PR DESCRIPTION
**Pull request by 9prady9**
Fixes #103

### CODE REVIEW

File: src/arith/mod.rs [hunk 1]
```diff
@@ -403,33 +393,51 @@ arith_scalar_spec!(i32);
...
-                }
+                add(&self, &rhs, false)
+            }
+        }
+
+        impl<'a> $op_name<&'a Array> for Array {
+            type Output = Array;
+
+            fn $fn_name(self, rhs: &'a Array) -> Array {
+                add(&self, rhs, false)
```
Comment (jramapuram) [inline src/arith/mod.rs]: The template always calls `add`? Did you mean to do something like `fn_name` ?

File: src/arith/mod.rs [hunk 2]
```diff
@@ -403,33 +393,51 @@ arith_scalar_spec!(i32);
...
+            }
+        }
+
+        impl<'a> $op_name<Array> for &'a Array {
+            type Output = Array;
+
+            fn $fn_name(self, rhs: Array) -> Array {
+                add(self, &rhs, false)
```
Comment (jramapuram) [inline src/arith/mod.rs]: Same as above comment. arith_func is used for all of these:

```rust
arith_func!(Add   , add   );
arith_func!(Sub   , sub   );
arith_func!(Mul   , mul   );
...
```

File: src/arith/mod.rs (hunk 1 above)
Comment (9prady9) [inline src/arith/mod.rs]: My bad, forgot to replace it with `$fn_name` after testing with `Add` trait.

File: src/arith/mod.rs (hunk 2 above)
Comment (9prady9) [inline src/arith/mod.rs]: Will fix. Thanks.

**Pull request merged**